extensions/parquet/table/src/main/java/io/deephaven/parquet/table/location/ParquetTableLocation.java

```diff
@@ -82,10 +82,11 @@ public ParquetTableLocation(@NotNull final TableKey tableKey,
         final ParquetMetadata parquetMetadata;
         // noinspection SynchronizationOnLocalVariableOrMethodParameter
         synchronized (tableLocationKey) {
+            // Following will create a new ParquetFileReader and read the file metadata if it does not exist
             parquetFileReader = tableLocationKey.getFileReader();
-            parquetMetadata = tableLocationKey.getMetadata();
-            rowGroupIndices = tableLocationKey.getRowGroupIndices();
         }
+        parquetMetadata = tableLocationKey.getMetadata();
```

Comment:
No reason to synchronize on only one. They are all internally synchronized. The reason for synchronizing on the TLK before calling these is just to minimize lock/unlock calls.